constraints/binRightmost.js

```diff
@@ -0,0 +1,60 @@
+/* Assign a violation for every node of category cat 
+such that its rightmost child of category (cat-1) 
+has more than two children.
+*/
+
+function binMaxRightmostBranches(s, ptree, cat) {
+  var vcount = 0;
+  //base case: we are at leaf && there are no children
+  //make sure there is children
+  if (ptree.children && ptree.children.length) {
+    if (ptree.cat === cat) {
+      //check rightmost child
+      var rightMost = ptree.children.length - 1;
+      var rightMostChild = ptree.children[rightMost];
+      if (rightMostChild.children && rightMostChild.children.length > 2) {
+        vcount++;
+      }       
+    }
+    //check other nodes in ptree
+    for(var i = 0; i < ptree.children.length; i++) {
+      vcount += binMaxRightmostBranches(s, ptree.children[i], cat);
+    }       
+  }
+  return vcount;
+};
+var parent_ptree;
```

Comment:
Define this variable inside the function so the function is self-contained (better code hygiene).